PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/StringSequenceNode.java

```diff
@@ -0,0 +1,165 @@
+package nl.tudelft.pl2016gr2.model;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+/**
+ * A simple implementation of <code>Node</code> that represents the DNA sequence as String.
+ *
+ * @author Wouter Smit
+ */
+public class StringSequenceNode extends AbstractNode {
+
+  private String sequence;
+  private HashSet<String> genomes;
```

Comment:
We could maybe create a `ArrayList<Integer, String>` and `HashMap<String,Integer>` in the graph, which store `<id, genome name>` (where the arraylist index is the id) and `<genome name, id>` pairs and store only the ids of the genomes in the nodes. This will reduce storage and speed up the comparison of genomes a lot (comparing numbers is a lot faster than comparing strings).
Then we could also give the phylogenetic tree a reference to these hashmaps and use the same IDs there.

I tested reading a graph and checking how many characters were actually used to store the genome names in all of the graph nodes. It was actually a lot higher than I expected:
for the graph of 328 nodes there are 62 782 632 characters used to store the genome names in the nodes of the tree!
So that's 62 782 632 * 2 = 120MB!

After this I have tried to determine the total amount of memory used by the graph by running this code:
```java
GfaReader reader = new GfaReader("TB328.gfa");
System.gc();
System.out.println("before = " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()));
OriginalGraph graph = reader.read();
System.gc();
System.out.println("after = " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()));
```
resulting in:
```
before = 2726888
after = 583088568
```
Se around 553MB was used by the graph.
> the results of multiple runs were pretty consistens, within ~200kb

Then I tried exactly the same code, but I commented out the `setGenomes()` call in the parser (so everything was the same, except that the genomes weren't set for the nodes). This resulted in:
```
before = 2726816
after = 156414328
```
So around 147MB was used by the graph.

This is a reduction of 73.5%!!
The genomes names are 17 characters (34 bytes) long, so if we use integers (4 bytes) as IDs instead of the actual strings, we should be able to save around `1 - 4 / 34 = 88.2%` of the space which is currently used by the genome names, resulting in a reduction of `73.5% * 88.2% = 64.9%` of the memory used by the graph.

Since the references to the string are coppied when new graphs are created (the strings are not duplicated) this will result in a bit lower memory reduction when the application is actually doing stuff (splitting/comparing graphs), but even if this would reduce the memory by only 10% (probably still around 30-40% though) this would be worth it in my opinion.
> someone could make an issue out of this